qualang_tools/addons/calibration/calibrations_API.py

```diff
@@ -0,0 +1,1114 @@
+        :return: None
+        """
+        calibration = "T1"
+        self._user_checks(calibration, scan_variables, iterations, cooldown_time)
+        if (len(scan_variables) > 1) or (scan_variables[0][0] != "duration"):
+            raise Exception("Only 'duration' scans are available for T1 calibration.")
+
+    def set_ramsey(
+        self, scan_variables, iterations=1, cooldown_time=None, idle_time=None
+    ):
+        """Performs a Ramsey-like measurement (pi/2 - idle time - pi/2).
+        The available scanning parameters are:
+        'duration' (idle time in clock cycles unit (4 ns)),
+        'frequency' (pulse frequency in Hz, must be an integer).
+
+        :param scan_variables: variables to be scanned. Must be an array of tuples [('variable', [values])]. 1D and 2D scans are available.
+        :param iterations: number of averaging iterations. Must be a positive Python integer.
+        :param cooldown_time: resonator or qubit cooldown time in clock cycles (4ns). Must be an integer.
+        :param idle_time: Ramsey sequence idle time in clock cycles unit (4 ns). Must be an integer larger than 4.
+        :return: None
+        """
+        calibration = "Ramsey"
+        self._user_checks(calibration, scan_variables, iterations, cooldown_time)
+        # Check idle time
+        if idle_time is None or (idle_time >= 4 and type(idle_time) is int):
+            self.user_specifics["idle_time"] = idle_time
+        elif cooldown_time == 0:
+            self.user_specifics["cooldown_time"] = None
+        else:
+            raise Exception(
+                "idle_time must be 0 or a python positive integer larger than 4."
+            )
+
+    def set_raw_traces(self, scan_variables=(), iterations=1, cooldown_time=None):
+        """Acquires the raw and averaged traces from the two inputs of the OPX.
+
+        :param scan_variables: None
+        :param iterations: number of averaging iterations. Must be a positive Python integer.
+        :param cooldown_time: resonator or qubit cooldown time in clock cycles (4ns). Must be an integer.
+        :return: None
+        """
+        calibration = "raw_traces"
+        self._user_checks(calibration, scan_variables, iterations, cooldown_time)
+
+    def set_time_of_flight(
+        self, scan_variables=(), iterations=1, cooldown_time=None, threshold=10 / 4096
+    ):
+        """Measures the time of flight and offsets of the two inputs of the OPX.
+
+        :param scan_variables: None
+        :param iterations: number of averaging iterations. Must be a positive Python integer.
+        :param cooldown_time: resonator or qubit cooldown time in clock cycles (4ns). Must be an integer.
+        :param threshold: detection threshold for time of flight estimation.
+        :return: None
+        """
+        calibration = "time_of_flight"
+        self._user_checks(calibration, scan_variables, iterations, cooldown_time)
+        self.user_specifics["threshold"] = threshold
```

Comment:
shouldn't this be part of the specific calibration?

Reply:
It is, it is only part of the time of flight calibration. I don't understand

Reply:
I think it should be: `self.user_specifics["time_of_flight"]["threshold"]`

Reply:
Oh I see, yes !